inc/types/class-base.php

```diff
@@ -152,4 +165,44 @@ protected function render_form( Client $client, WP_Error $errors = null ) {
 	protected function get_nonce_action( Client $client ) {
 		return sprintf( 'oauth2_authorize:%s', $client->get_id() );
 	}
+
+	/**
+	 * Get and validate PKCE parameters from a request.
+	 *
+	 * @return string[] code_challenge and code_challenge_method
+	 */
+	private function handle_pkce() {
+		$code_challenge        	= isset( $_GET['code_challenge'] ) ? wp_unslash( $_GET['code_challenge'] ) : null;
+		$code_challenge_method 	= isset( $_GET['code_challenge_method'] ) ? wp_unslash( $_GET['code_challenge_method'] ) : null;
```

Comment:
Can we instead pass `wp_unslash( $_GET )` as a parameter into this function to separate the globals out?